x-pack/plugins/security_solution/public/overview/containers/overview_risky_host_links/use_hosts_risk_score.ts

```diff
@@ -96,21 +96,27 @@ export const useRiskyHostLinks = ({ to, from }: { to: string; from: string }) =>
   }, [addError, error, setIsModuleEnabled]);
 
   useEffect(() => {
-    start({
-      data,
-      timerange: { to, from, interval: '' },
-      defaultIndex: [RISKY_HOSTS_INDEX],
-      filterQuery: '',
-    });
-  }, [start, data, to, from]);
+    if (riskyHostsFeatureEnabled && (hostName || timerange)) {
+      start({
+        data,
+        timerange: timerange ? { to: timerange.to, from: timerange.from, interval: '' } : undefined,
+        hostName,
+        defaultIndex: [HOST_RISK_SCORES_INDEX],
+      });
+    }
+  }, [start, data, timerange, hostName, riskyHostsFeatureEnabled]);
+
+  if ((!hostName && !timerange) || !riskyHostsFeatureEnabled) {
+    return null;
+  }
+
+  const hits = result?.rawResponse?.hits?.hits;
 
   return {
-    listItems: isRiskyHostHit(result?.rawResponse?.hits?.hits?.[0]?._source)
-      ? getListItemsFromHits(
-          result?.rawResponse?.hits?.hits?.map((hit) => hit._source) as RiskyHost[]
-        )
+    result: isHostsRiskScoreHit(hits?.[0]?._source)
+      ? (hits?.map((hit) => hit._source) as HostsRiskScore[])
       : [],
     isModuleEnabled,
-    loading,
+    loading: isModuleEnabled === undefined ? true : loading,
```

Comment:
aaaaaaaaat???

Reply:
we could also do `loading: isModuleEnabled ? loading : true`